benchmarks/templates/benchmarks/table.html

```diff
@@ -99,7 +99,11 @@
                     </div>
                     <div class="submitter">
                         {% if model.user.display_name is not None %}
-                            {{ model.user.display_name }}
+                            {% if not model.public and model.competition is not None %}
+                                Anonymous Submitter
+                            {% else %}
+                                {{ model.user.display_name }}
+                            {% endif %}
```

Comment:
let's also keep the model name anonymous, e.g. "Anonymous submission"

then I think we can ignore the user line and just put `&nbsp;` (instead of the current "Anonymous Submitter")